cmd/agent/app/agent.go

```diff
@@ -112,20 +112,24 @@ func run(ctx context.Context, karmadaConfig karmadactl.KarmadaConfig, opts *opti
 		return fmt.Errorf("error building kubeconfig of member cluster: %w", err)
 	}
 	clusterKubeClient := kubeclientset.NewForConfigOrDie(clusterConfig)
-	// ensure namespace where the impersonation secret to be stored in member cluster.
-	if _, err = util.EnsureNamespaceExist(clusterKubeClient, opts.ClusterNamespace, false); err != nil {
-		return err
-	}
+	controlPlaneKubeClient := kubeclientset.NewForConfigOrDie(controlPlaneRestConfig)
 
-	// create a ServiceAccount for impersonator in cluster.
-	impersonationSA := &corev1.ServiceAccount{}
-	impersonationSA.Namespace = opts.ClusterNamespace
-	impersonationSA.Name = names.GenerateServiceAccountName("impersonator")
-	if _, err = util.EnsureServiceAccountExist(clusterKubeClient, impersonationSA, false); err != nil {
+	registerOption := util.RegistryClusterOption{
+		ClusterNamespace:   opts.ClusterNamespace,
+		ClusterName:        opts.ClusterName,
+		ReportSecrets:      opts.ReportSecrets,
+		ClusterAPIEndpoint: opts.ClusterAPIEndpoint,
+		ProxyServerAddress: opts.ProxyServerAddress,
+		ClusterProvider:    opts.ClusterProvider,
+		ClusterRegion:      opts.ClusterRegion,
+		ClusterZone:        opts.ClusterZone,
+		DryRun:             false,
+	}
+	clusterSecret, impersonatorSecret, err := util.ObtainCredentialsFromMemberCluster(clusterKubeClient, registerOption)
+	if err != nil {
```

Comment:
How about obtaining secrets directly in the method `RegisterClusterInControllerPlane`?

Reply:
> How about obtaining secrets directly in the method RegisterClusterInControllerPlane?

obtain secret is from member cluster,  registry cluster in host cluster; logic  separated may be better?

Reply:
IMOP, the input parameters of the func `RegisterClusterInControllerPlane` is a little bit large, so I suggest this. Perhaps we can put them into `registerOption`.

Reply:
> so I suggest this. Perhaps we can put them into registerOption
good idea